patches/components-vector_icons-aggregate_vector_icons.py.patch

```diff
@@ -11,28 +31,51 @@ index 3e5a94a7f053ffc66f1a820073265cbfe1332b39..0175ec714b441df5e8fa58e9391e2de6
    """Compiles all .icon files in a directory into two C++ files.
  
    Args:
-@@ -111,6 +111,8 @@ def AggregateVectorIcons(working_directory, file_list, output_cc, output_h):
+@@ -111,8 +124,12 @@ def AggregateVectorIcons(working_directory, file_list, output_cc, output_h):
        file_list: A file containing the list of vector icon files to process.
        output_cc: The path that should be used to write the .cc file.
        output_h: The path that should be used to write the .h file.
 +      alt_working_directory: The alternative path to the directory that holds
 +          the .icon files
    """
  
++  leo_overrides = get_overrides()
++
    # For each file in |file_list|, place it in |path_map| if its extension is
-@@ -129,6 +131,11 @@ def AggregateVectorIcons(working_directory, file_list, output_cc, output_h):
+   # .icon. This will map the icon's name to its path, e.g.,
+   # path_map['cat'] = 'foo/bar/cat.icon'.
+@@ -129,6 +146,14 @@ def AggregateVectorIcons(working_directory, file_list, output_cc, output_h):
      if extension != ".icon":
        Error("Only filenames " + icon_name + ".icon are allowed.")
  
 +    # Check for alternative path
 +    alt_icon_path = os.path.join(alt_working_directory, os.path.basename(icon_path))
-+    if (os.path.exists(alt_icon_path)):
++    
++    if icon_name in leo_overrides:
++      icon_path = leo_overrides[icon_name]
++    elif os.path.exists(alt_icon_path):
 +      icon_path = alt_icon_path
 +
```

Comment:
I know you are just updating this, but it would be ideal to externalize the bulk of this as well.